### PR TITLE
[TT-5371] OAS patch endpoint

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description
Added OAS patch endpoint to configure with simple OAS and query params

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2365640177)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: f7821d5456c675d634e85647dabed83cd1e59e52 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2365645223)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: 4214baf90669bfb80db4eda72c93923c41ea0a9d 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2365704197)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: 6187d8a1a21b110d3ba188be16c6ca89f8f9f226 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2366365792)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: 63df39bdf7ee9cba9f40538e59165f551ec7319d 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2366394028)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: ac44c7c91b64285f8bd4e0881c811324d37633c5 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2368890663)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: 6d17318886a72e836b2a3300a345d4743f82c934 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2369393409)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: 11be289c2ebb616ea1c84a28ec2ea424b4c95ae7 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2370071912)

**Review by urbanishimwe — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: ab73b1995fd25023d62c416c06f66eeabdafd726 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2370172338)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: 77b30e1d9ae2b6096a86effb81499883d0764bb1 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2370345513)

File: rpc/rpc_client.go [hunk 1]
```diff
@@ -3,14 +3,15 @@ package rpc
...
 	"errors"
-	"github.com/TykTechnologies/tyk-pump/serializer"
 	"net"
 	"strconv"
 	"sync"
 	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/TykTechnologies/tyk-pump/serializer"
```
Comment (jeffy-mathew) [inline rpc/rpc_client.go:13]: `goimports` reported error on this. nothing to do with the PR

File: gateway/api.go [hunk 2]
```diff
@@ -61,6 +62,10 @@ import (
 	gql "github.com/jensneuse/graphql-go-tools/pkg/graphql"
 )
 
+var (
+	ErrRequestMalformed = errors.New("Request malformed")
```
Comment (furkansenharputlu) [inline gateway/api.go:66]: Lower case in errors?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4062/merge
Commit: 9e828fc7b0c786eafde55e36f45ede47344cd5af 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2371479256)

**Review by lghiur — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 8625c15d8a3dd5f14497014de2b193a8f0b723cf [TT-5371] OAS patch endpoint (#4062)

* Add OAS PATCH endpoint

[changelog]
added: OAS patch endpoint.
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2371963377)

**Pull request merged**